src/wallet/wallet.cpp

```diff
@@ -3913,6 +3944,10 @@ CWallet* CWallet::CreateWalletFromFile(const std::string walletFile)
         if (walletdb.ReadBestBlock(locator))
             pindexRescan = FindForkInGlobalIndex(chainActive, locator);
     }
+
+    walletInstance->m_last_block_processed = chainActive.Tip();
```

Comment:
Doesn't this entire function need cs_main ?

Reply:
It would appear so (though that isn't introduced here)....I'll let @practicalswift take a look since that appears to be his project atm.

Reply:
Yes, according to my annotations (#11226) calling `CreateWalletFromFile` requires holding `cs_main`:

```
src/wallet/wallet.h:    static CWallet* CreateWalletFromFile(const std::string walletFile)
                            EXCLUSIVE_LOCKS_REQUIRED(cs_main);
```

This is due to:
* `CreateWalletFromFile` is reading the variable `chainActive` which requires holding the mutex `cs_main`.
* `CreateWalletFromFile` calls `FindForkInGlobalIndex` which reads the variable `mapBlockIndex`. Reading the variable `mapBlockIndex` requires holding the mutex `cs_main`.